api/types/system/info.go

```diff
@@ -73,6 +73,7 @@ type Info struct {
 	SecurityOptions     []string
 	ProductLicense      string               `json:",omitempty"`
 	DefaultAddressPools []NetworkAddressPool `json:",omitempty"`
+	CDISpecDirs         []string
```

Comment:
Can you also add the new field 

- add the new field to the swagger https://github.com/moby/moby/blob/69c9adb7d3868cb0560b1ffcef798015c5a70510/api/swagger.yaml#L4860
- add a bullet to the API version-history https://github.com/moby/moby/blob/69c9adb7d3868cb0560b1ffcef798015c5a70510/docs/api/version-history.md#v144-api-changes

For both, we probably should mention that it requires the daemon to have experimental enabled (for this release)

Reply:
Also wondering if we still need a`omitempty` and to omit the field for API < 1.44 (as this fields "doesn't exist" in older API versions).

We're generally more relaxed in returning additional fields (as we do maintain an "open schema", so happy to hear from others on that.

Reply:
> Also wondering if we still need a`omitempty` and to omit the field for API < 1.44 (as this fields "doesn't exist" in older API versions).

I sure hope not. I wouldn't want to juggle multiple versioned `Info` structs if it can be avoided. [The `omitempty` tag suppresses marshaling of empty, non-nil slices.](https://go.dev/play/p/6etn-kZ7aI5)

Reply:
> For both, we probably should mention that it requires the daemon to have experimental enabled (for this release)

This got me thinking. Should we also set `CDISpecDirs` to nil explicitly if `experimental != true`?

Reply:
I like that idea; users could then use `len(Info.CDISpecDirs) > 0` as the signal that CDI support is enabled on the daemon, without needing to know whether the feature has graduated from experimental.

Reply:
I have updated the implementation and tests to clear the setting if `experimental != true`. I have not removed the gating of the call to `RegisterCDIDriver` for the time being but can if anyone feels strongly about it. The downside is that we would have to make two changes once we graduate this to a non-experimental feature.